packages/browser-utils/src/metrics/inp.ts

```diff
@@ -0,0 +1,152 @@
+  drag: 'drag',
+  dragenter: 'drag',
+  dragleave: 'drag',
+  dragover: 'drag',
+  drop: 'drag',
+  keydown: 'press',
+  keyup: 'press',
+  keypress: 'press',
+  input: 'press',
+};
+
+/** Starts tracking the Interaction to Next Paint on the current page. */
+function _trackINP(): () => void {
+  return addInpInstrumentationHandler(({ metric }) => {
+    const client = getClient();
+    if (!client || metric.value == undefined) {
+      return;
+    }
+
+    const entry = metric.entries.find(entry => entry.duration === metric.value && INP_ENTRY_MAP[entry.name]);
+
+    if (!entry) {
+      return;
+    }
+
+    const interactionType = INP_ENTRY_MAP[entry.name];
+
+    const options = client.getOptions();
+    /** Build the INP span, create an envelope from the span, and then send the envelope */
+    const startTime = msToSec((browserPerformanceTimeOrigin as number) + entry.startTime);
+    const duration = msToSec(metric.value);
+    const scope = getCurrentScope();
+    const activeSpan = getActiveSpan();
+    const rootSpan = activeSpan ? getRootSpan(activeSpan) : undefined;
+
+    const routeName = rootSpan ? spanToJSON(rootSpan).description : undefined;
+    const user = scope.getUser();
+
+    // We need to get the replay, user, and activeTransaction from the current scope
+    // so that we can associate replay id, profile id, and a user display to the span
+    const replay = client.getIntegrationByName<Integration & { getReplayId: () => string }>('Replay');
+
+    const replayId = replay && replay.getReplayId();
+
+    const userDisplay = user !== undefined ? user.email || user.id || user.ip_address : undefined;
+    const profileId = scope.getScopeData().contexts?.profile?.profile_id as string | undefined;
+
+    const name = htmlTreeAsString(entry.target);
+    const parentSampled = activeSpan ? spanIsSampled(activeSpan) : undefined;
+    const attributes: SpanAttributes = dropUndefinedKeys({
+      release: options.release,
+      environment: options.environment,
+      transaction: routeName,
+      [SEMANTIC_ATTRIBUTE_EXCLUSIVE_TIME]: metric.value,
+      user: userDisplay || undefined,
+      profile_id: profileId || undefined,
+      replay_id: replayId || undefined,
+    });
+
+    /** Check to see if the span should be sampled */
+    const [sampled] = sampleSpan(options, {
+      name,
+      parentSampled,
+      attributes,
+      transactionContext: {
+        name,
+        parentSampled,
+      },
+    });
+
+    // Nothing to do
+    if (!sampled) {
+      return;
+    }
+
+    const span = new SentrySpan({
+      startTimestamp: startTime,
+      endTimestamp: startTime + duration,
+      op: `ui.interaction.${interactionType}`,
+      name,
+      attributes,
+    });
+
+    span.addEvent('inp', {
+      [SEMANTIC_ATTRIBUTE_SENTRY_MEASUREMENT_UNIT]: 'millisecond',
+      [SEMANTIC_ATTRIBUTE_SENTRY_MEASUREMENT_VALUE]: metric.value,
+    });
+
+    const envelope = span ? createSpanEnvelope([span]) : undefined;
```

Comment:
span should always be defined here, no need to check for existence:

```suggestion
    const envelope = createSpanEnvelope([span]);
```